app/views/notifications/shared/_article_preview.html.erb

```diff
@@ -21,14 +21,25 @@
       </button>
 
       <button
-        class="ml-auto crayons-btn crayons-btn--ghost crayons-btn--icon-right crayons-btn--s reaction-readinglist reaction-button readinglist-button <%= Reaction.cached_any_reactions_for?(notification.mocked_object("article"), current_user, "readinglist") ? "reacted" : "" %>"
+        class="crayons-btn crayons-btn--ghost crayons-btn--icon-right crayons-btn--s reaction-readinglist reaction-button readinglist-button <%= Reaction.cached_any_reactions_for?(notification.mocked_object("article"), current_user, "readinglist") ? "reacted" : "" %>"
         data-reactable-id="<%= json_data["article"]["id"] %>"
         data-category="readinglist"
         data-reactable-type="Article">
         <span class="reaction-button-text hidden m:inline-block"><%= t("views.reactions.readingList.text") %></span>
         <%= crayons_icon_tag("small-save", class: "reaction-icon not-reacted", title: t("views.reactions.readingList.icon")) %>
         <%= crayons_icon_tag("small-save-filled", class: "reaction-icon--readinglist reaction-icon reacted", title: t("views.reactions.readingList.icon")) %>
       </button>
+
+      <% article = Article.find(json_data["article"]["id"]) %>
+      <% @notification = NotificationSubscription.where(notifiable_type: "Article", user_id: current_user.id, notifiable_id: article.id).first %>
```

Comment:
Is this lookup (and the ones in _comment_box) best performed here or in the controller? I think I've seen more cache-flavored calls to model methods in the views, not as many direct database calls